hack/ci/windows.ps1

```diff
@@ -418,7 +427,12 @@ Try {
 
     # Redirect to a temporary location. 
     $TEMPORIG=$env:TEMP
-    $env:TEMP="$env:TESTRUN_DRIVE`:\$env:TESTRUN_SUBDIR\CI-$COMMITHASH"
+    if ($null -eq $env:BUILD_NUMBER) {
+      $env:TEMP="$env:TESTRUN_DRIVE`:\$env:TESTRUN_SUBDIR\CI-$COMMITHASH"
+    } else {
+      # individual temporary location per CI build (even for a rebuild in CI without a new commithash)
+      $env:TEMP="$env:TESTRUN_DRIVE`:\$env:TESTRUN_SUBDIR\CI-$env:BUILD_NUMBER-$COMMITHASH"
```

Comment:
Is `BUILD_NUMBER` globally unique? Or is that the incrementing number per build? (`PR-123, build 1 .. 2 ...3`?)

Wondering if we should use `PR-<pr-number>\<build-number>` instead (that way we can associate each run with the PR it belongs to, without using the `$COMMITHASH`, as it may change when force-pushing)

Reply:
I was thinking about situations where two different PR's land on the same machine and both have (eg) "job 1".

The commit-sha helps to still make them unique (but is not easy to relate to a specific PR)